lib/blather/client/client.rb

```diff
@@ -229,9 +237,11 @@ def setup(jid, password, host = nil, port = nil, certs = nil, connect_timeout =
     # @private
     def handler_queue
       return if queue_size == 0
-      @handler_queue ||= GirlFriday::WorkQueue.new :handle_stanza, :size => queue_size do |stanza|
-        handle_data stanza
-      end
+      return @handler_queue if @handler_queue
+      @handler_queue = Class.new(Job)
```

Comment:
I was trying to keep it working the way you had it, otherwise the queue size would apply across all queues. You can't seem to create an instance of a queue in any other way with sucker_punch.